src/parser/mod.rs

```diff
@@ -11141,19 +11154,34 @@ impl<'a> Parser<'a> {
     /// FIRST_VALUE(x IGNORE NULL);
     /// ```
     fn parse_function_argument_list(&mut self) -> Result<FunctionArgumentList, ParserError> {
+        let mut clauses = vec![];
+
+        if dialect_of!(self is MsSqlDialect) {
+            // JSON_ARRAY(NULL ON NULL)
+            if self.parse_keywords(&[Keyword::NULL, Keyword::ON, Keyword::NULL]) {
+                clauses.push(FunctionArgumentClause::JsonNullClause(
+                    JsonNullClause::NullOnNull,
+                ));
+            }
+            // JSON_ARRAY(ABSENT ON NULL)
```

Comment:
Ah noticed we have this code few lines below as well, so that the above comment applies to that one as well. Actually for this line, I'm not sure we need the changes here at all, since if we've hit `RParen` we don't have any clauses to parse (so that the one below I imagine should be sufficient)?

Reply:
The first few lines of  `parse_keywords` are for `JSON_ARRAY(<json-null-clause>)`/``JSON_OBJECT(<json-null-clause>)`` which does not have any arguments only includes `json-null-clause` because T-SQL supports it.

playground:
<https://sqliteonline.com/#sqltext=%23url-sqlite%3Ddb-mssql%0D%0A%23tab-name%3DMS%20SQL%0D%0ASELECT%20JSON_ARRAY(NULL%20ON%20NULL)%2CJSON_OBJECT(ABSENT%20ON%20NULL)>

Reply:
hmm yeah that was my understanding, but not sure I follow that the change is needed to handle that scenario, in that neither of the [other clauses](https://github.com/apache/datafusion-sqlparser-rs/blob/43b0652bb1469e253ef3e1d9e606f43c30dba845/src/parser/mod.rs#L11182-L11224) should kick in or?

Reply:
Maybe this is a useful scenario?

```sql
SELECT JSON_OBJECT($json_param ABSENT ON NULL);
```

`$json_param` can be replaced to empty by user input.